build.gradle.kts

```diff
@@ -8,29 +31,31 @@ repositories {
 }
 
 dependencies {
-    implementation("org.junit.jupiter:junit-jupiter-api:5.8.1")
-    implementation("org.junit.jupiter:junit-jupiter-engine:5.8.1")
+    testImplementation(platform("org.junit:junit-bom:5.8.2"))
+    testImplementation("org.junit.jupiter:junit-jupiter")
 }
 
-tasks.test {
-    useJUnitPlatform()
+tasks.withType<Test> {
+    useJUnitPlatform() // JUnit 5
 }
 
 version = "0.1.1"
 group = "com.github.zeckie"
 
-// Work out current java version
+/*
+* Work out current java version
+* Note: versions prior to JEP223 (Java 9) had version numbers starting with "1." so
+* will be counted as major version 1.
+*/
+@Suppress("PropertyName") // IDEA doesn't recognise MIN_JAVA_VER as a constant
 val MIN_JAVA_VER = 9
-val fullVersion = System.getProperty("java.version")
-val majorVersion = Integer.parseInt(fullVersion.substring(0, fullVersion.indexOf('.')))
-System.out.println("Java major version: " + majorVersion)
+val fullVersion: String = System.getProperty("java.version")
+val matchMajor: Matcher = Pattern.compile("^\\d+").matcher(fullVersion)
+val majorVersion = if (matchMajor.find()) Integer.parseInt(matchMajor.group()) else 0
```

Comment:
As the programming language being uses here is Kotlin, this should be written with more idiomatic Kotlin:
```kt
val fullVersion: String = System.getProperty("java.version")
val majorVersion = Regex("^\\d+").find(fullVersion)?.value.toInt() ?: 0
// or
val majorVersion = fullVersion.split("\\D".toRegex())[0].toIntOrNull() ?: 0
```

It would actually be much better to use string functions for this though:
```kt
val majorVersion = fullVersion.takeWhile { it.isDigit() }
// or
val majorVersion = fullVersion.takeWhile(Char::isDigit)
```

You can learn about Kotlin's `text` package which includes `String` and `Regex` functionality [here](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/).
And [here](https://www.youtube.com/watch?v=n4WBip822A8) is a video tutorial if you prefer.

Regardless of all that, it should really utilize Gradle's [`JavaVersion` api](https://docs.gradle.org/current/javadoc/org/gradle/api/JavaVersion.html), which has all of this functionality built in, so theres no need to reinvent the wheel.